# ***Taller 2***

Nombre: Jazmin Muñoz Astorga

Profesor: Francisco Alfaro

Ayudantes: Krischna y Karen

Para responder las preguntas siguientes, utilice la base de datos ceosal1 de la librería wooldridge.
Asegúrese de cargarla correctamente (sin nulos) y de especificar el programa de forma rigurosa para
evitar cambios por aleatoriedad.

**1. (10 puntos) Examine la base y elabore una breve descripción de sus variables, distinguiendo
entre dependientes, explicativas, continuas y dummies. ¿Para qué tipo de estudio económico
se emplea habitualmente esta base de datos?**  


In [12]:
install.packages("wooldridge")
library(wooldridge)
set.seed(1)
data(ceosal1)
ceosal1 <- na.omit(ceosal1)
str(ceosal1)


Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



'data.frame':	209 obs. of  12 variables:
 $ salary  : int  1095 1001 1122 578 1368 1145 1078 1094 1237 833 ...
 $ pcsalary: int  20 32 9 -9 7 5 10 7 16 5 ...
 $ sales   : num  27595 9958 6126 16246 21783 ...
 $ roe     : num  14.1 10.9 23.5 5.9 13.8 ...
 $ pcroe   : num  106.4 -30.6 -16.3 -25.7 -3 ...
 $ ros     : int  191 13 14 -21 56 55 62 44 37 37 ...
 $ indus   : int  1 1 1 1 1 1 1 1 1 1 ...
 $ finance : int  0 0 0 0 0 0 0 0 0 0 ...
 $ consprod: int  0 0 0 0 0 0 0 0 0 0 ...
 $ utility : int  0 0 0 0 0 0 0 0 0 0 ...
 $ lsalary : num  7 6.91 7.02 6.36 7.22 ...
 $ lsales  : num  10.23 9.21 8.72 9.7 9.99 ...
 - attr(*, "time.stamp")= chr "25 Jun 2011 23:03"




Descripción de variables:

Dependiente:

salary: salario anual del CEO

Explicativas continuas:

sales: ventas anuales de la empresa.

roe: retorno sobre el capital.

ros: retorno sobre ventas.

Explicativas dummy:

indus, finance, consprod, utility: sectores industriales (industria, finanzas, productos de consumo, servicios públicos).

grad: si el CEO tiene postgrado.

college: si el CEO tiene título universitario.

Uso habitual:
Se emplea para estudiar la determinación del salario de los CEOs, analizando el impacto del desempeño financiero de la empresa (roe, ros, sales) y características del sector o del CEO.

**2. (5 puntos) Genere una variable indicadora (dummy) que tome el valor 1 cuando la variable
sales supere su promedio y 0 en caso contrario. Asigne a esta nueva variable un nombre que
sea fácilmente identificable.**


In [20]:
media_sales <- mean(ceosal1$sales)
ceosal1$sales_mayor_promedio <- ifelse(ceosal1$sales > media_sales, 1, 0)

# Verificar que se creó bien
table(ceosal1$sales_mayor_promedio)


  0   1 
155  54 

Se creó la variable sales_mayor_promedio, que es una dummy igual a 1 si sales > mean(sales) y 0 en caso contrario. Este nombre es fácilmente identificable porque indica directamente que la empresa tiene ventas superiores al promedio.

**3. (5 puntos) Cree una variable de la multuplicación entre la variable dummy antes creada y la
variable indus. ¿Qué indica esta interacción?**


In [22]:
ceosal1$sales_mayor_promedio_indus <- ceosal1$sales_mayor_promedio * ceosal1$indus
head(ceosal1[, c("sales", "sales_mayor_promedio", "indus", "sales_mayor_promedio_indus")])

,sales,sales_mayor_promedio,indus,sales_mayor_promedio_indus
,<dbl>,<dbl>,<int>,<dbl>
1,27595.0,1,1,1
2,9958.0,1,1,1
3,6125.9,0,1,0
4,16246.0,1,1,1
5,21783.2,1,1,1
6,6021.4,0,1,0


La interacción sales_mayor_promedio_indus indica si existe un efecto adicional (sinérgico o sustractivo) en el salario del CEO cuando la empresa pertenece al sector industrial y además tiene ventas superiores al promedio, más allá de los efectos individuales de cada característica por separado.

**4. (5 puntos) Cree una variable de la multiplicación entre la variable continua roe y la variable
dummy antes creada. ¿Qué indica esta interacción?**


In [25]:
ceosal1$sales_mayor_promedio_roe <- ceosal1$sales_mayor_promedio*ceosal1$roe
head(ceosal1[, c("sales", "sales_mayor_promedio", "indus", "sales_mayor_promedio_indus", "roe", "sales_mayor_promedio_roe")], 10)

,sales,sales_mayor_promedio,indus,sales_mayor_promedio_indus,roe,sales_mayor_promedio_roe
,<dbl>,<dbl>,<int>,<dbl>,<dbl>,<dbl>
1,27595.0,1,1,1,14.1,14.1
2,9958.0,1,1,1,10.9,10.9
3,6125.9,0,1,0,23.5,0.0
4,16246.0,1,1,1,5.9,5.9
5,21783.2,1,1,1,13.8,13.8
6,6021.4,0,1,0,20.0,0.0
7,2266.7,0,1,0,16.4,0.0
8,2966.8,0,1,0,16.3,0.0
9,4570.2,0,1,0,10.5,0.0


La interacción sales_mayor_promedio_roe indica si el efecto del retorno sobre el capital (ROE) en el salario del CEO varía dependiendo de si la empresa tiene ventas superiores al promedio. Permite evaluar si el desempeño financiero es más (o menos) valorado en empresas con ventas altas.

**5. (5 puntos) Cree una variable que indique si el compañia corresponde a industria (indus) ó si
finance. Asigne a esta nueva variable un nombre que sea fácilmente identificable.**

In [33]:
ceosal1$indus_o_finance <- ceosal1$indus + ceosal1$finance
table(ceosal1$indus, ceosal1$finance, ceosal1$indus_o_finance)

, ,  = 0

   
     0  1
  0 96  0
  1  0  0

, ,  = 1

   
     0  1
  0  0 46
  1 67  0


La variable indus_o_finance se creó como la suma de indus y finance. En esta base de datos, no existen empresas que pertenezcan simultáneamente a ambos sectores, por lo que la variable toma valor 0 si la empresa no es de industria ni finance (96 observaciones), y valor 1 si pertenece a industria o a finance (113 observaciones en total: 46 de finance y 67 de industria). El nombre es fácilmente identificable porque combina "indus" y "finance" indicando que agrupa ambos sectores.

***2 Modelos de Regresión Lineal (70 puntos)***

**1. (15 puntos) Genere un modelo de regresión lineal, sobre alguna de las variables dependientes,
con respecto a las variables sales, roe y ros. Interprete sus resultados y la significancia de los
regresores. ¿El modelo tiene significancia global al 5%?**


In [34]:
modelo <- lm( salary ~ sales + roe + ros , data = ceosal1 )
summary ( modelo )


Call:
lm(formula = salary ~ sales + roe + ros, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
-1495.0  -486.5  -242.8    98.9 13516.4 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept) 864.117516 228.399737   3.783 0.000203 ***
sales         0.015482   0.008954   1.729 0.085291 .  
roe          22.003312  11.516554   1.911 0.057454 .  
ros          -1.105191   1.450240  -0.762 0.446892    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1360 on 205 degrees of freedom
Multiple R-squared:  0.03191,	Adjusted R-squared:  0.01775 
F-statistic: 2.253 on 3 and 205 DF,  p-value: 0.08337


En este modelo estimé el salario del CEO en función de las ventas, el ROE y el ROS. Los resultados muestran que ninguna variable es significativa al 5%, aunque las ventas y el ROE lo son al 10% con p-values de 0.085 y 0.057 respectivamente. El ROS no es significativo ni siquiera al 10% con un p-value de 0.447. Por cada millón de dólares adicional en ventas, el salario aumenta solo
15.48 dolares; por cada punto mas de ROE, el salario sube
22,003 dólares; y por cada punto más de ROS, el salario baja 1,105 dólares, pero esto último no es confiable estadísticamente. El modelo tiene un R-cuadrado de 0.0319, o sea, explica apenas el 3.19% de la variabilidad del salario. El test F global da un p-value de 0.08337, que es mayor a 0.05, por lo tanto el modelo no es globalmente significativo al 5%. En conclusión, estas variables por sí solas no logran explicar bien las diferencias en los salarios de los CEOs.

**2. (10 puntos) Al modelo de regresión lineal anterior, agregue las variables creadas en el inciso 1.2
y 1.4, interprete sus resultados. ¿Cómo afectan los regresores de estas variables en el salario?**

In [36]:
modelo <- lm( salary ~ sales + roe + ros + sales_mayor_promedio + sales_mayor_promedio_roe , data = ceosal1 )
summary ( modelo )



Call:
lm(formula = salary ~ sales + roe + ros + sales_mayor_promedio + 
    sales_mayor_promedio_roe, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
 -976.0  -469.1  -263.4    97.2 13566.3 

Coefficients:
                           Estimate Std. Error t value Pr(>|t|)   
(Intercept)              783.298711 258.064395   3.035  0.00272 **
sales                      0.006657   0.011197   0.594  0.55285   
roe                       24.472338  13.088222   1.870  0.06295 . 
ros                       -0.986904   1.457818  -0.677  0.49919   
sales_mayor_promedio     513.083213 510.134319   1.006  0.31572   
sales_mayor_promedio_roe  -9.498301  25.587585  -0.371  0.71087   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1361 on 203 degrees of freedom
Multiple R-squared:  0.04059,	Adjusted R-squared:  0.01696 
F-statistic: 1.718 on 5 and 203 DF,  p-value: 0.132


En este modelo agregue al anterior la dummy de ventas sobre el promedio y su interaccion con el ROE. Los resultados muestran que ninguna de las nuevas variables es significativa al 5%, con p-values de 0.315 y 0.710 respectivamente. Al incluir estas variables, las ventas dejan de ser significativas y solo el ROE se mantiene significativo al 10%. La variable sales_mayor_promedio tiene un coeficiente de 513.08, lo que sugiere que las empresas con ventas altas pagan en promedio 513,083 mas a sus CEOs, aunque este efecto no es estadisticamente confiable para empresas con ventas altas, un punto mas de ROE aumenta el salario en 14,97 miles de dolares , mientras que para empresas con ventas bajas el aumento es de 24.47 miles de dolares. Esto sugiere que el ROE es mejor recompensado en empresas mas pequeñas. El R-cuadrado subio muy poco, de 0.0319 a 0.0406, confirmando que estas variables no mejoran el modelo.

**3. (15 puntos) Realiza un test de hipotesis para evaluar si las dos variables que se agregaron son
significativas para el modelo simultaneamente. Use significancia del 5%**

In [38]:
F_critico <- qf (0.95 , 2 , 203)
R2_r <- 0.03191
R2_ur <- 0.04059
F_est <- ( R2_ur - R2_r )/(1 - R2_ur )* 203 /2
if ( F_est >F_critico ) {"Se rechaza H0"} else {"No se rechaza H0"}

[1] "No se rechaza H0"

Dado que el valor crítico al 5% con 2 y 203 grados de libertad es 3.04, y 0.918 < 3.04, no se rechaza la hipótesis nula. Por lo tanto, las dos variables agregadas no son conjuntamente significativas para explicar el salario del CEO.

**4. (15 puntos) Volviendo a la regresión del salario en función de sales, roe y ros. Considere
dos grupos, Grupo 1: Aquellos clientes que sales es mayor al promedio, y Grupo 2: Aquellos
clientes que sales es menor al promedio. Realice un test de hipotesis para grupos al 5% de
significancia. Use el metodo que más le acomode.**

In [40]:
ceosal1$sales_D <- ceosal1$sales * ceosal1$sales_mayor_promedio
ceosal1$roe_D <- ceosal1$roe * ceosal1$sales_mayor_promedio
ceosal1$ros_D <- ceosal1$ros * ceosal1 $sales_mayor_promedio

modelo <- lm( salary ~ sales + roe + ros + sales_mayor_promedio + sales_D+ roe_D+ ros_D, data = ceosal1 )
summary ( modelo )
F_critico <- qf (0.95 , 4 ,201)
R2_ur <- 0.06028
R2_r <- 0.03191
F_est <- ( R2_ur - R2_r )/(1 - R2_ur )* 201 /4
if ( F_est > F_critico ) {"Se rechaza H0"} else {"No se rechaza H0"}



Call:
lm(formula = salary ~ sales + roe + ros + sales_mayor_promedio + 
    sales_D + roe_D + ros_D, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
-1024.7  -456.7  -210.2    83.2 13712.4 

Coefficients:
                       Estimate Std. Error t value Pr(>|t|)  
(Intercept)           331.86179  352.76387   0.941   0.3480  
sales                   0.13362    0.06635   2.014   0.0454 *
roe                    25.63426   13.07068   1.961   0.0512 .
ros                    -0.47570    1.66943  -0.285   0.7760  
sales_mayor_promedio 1043.02106  578.05672   1.804   0.0727 .
sales_D                -0.13033    0.06731  -1.936   0.0542 .
roe_D                 -17.86857   28.02885  -0.638   0.5245  
ros_D                   1.55160    3.84565   0.403   0.6870  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1353 on 201 degrees of freedom
Multiple R-squared:  0.06028,	Adjusted R-squared:  0.02756 
F-statistic: 1.842 on 7 and 201

[1] "No se rechaza H0"

El resultado esperado es no rechazar la hipótesis nula, lo que indicaría que no existen diferencias estadísticamente significativas entre ambos grupos, es decir, el efecto de las ventas, el ROE y el ROS sobre el salario del CEO es similar independientemente de si la empresa tiene ventas altas o bajas.

**5. (15 puntos) Usando la regresión del salario en función de sales, roe y ros, se distinguen los
grupos mediante las dummies; indus, finance, consprod y utility, que corresponden al tipo de
empresa. Realice mediante test de Chow, con 5% de significancia, si los modelos para cada
grupo son iguales.**

In [44]:
ceosal1_indus <- subset ( ceosal1 , indus ==1)
ceosal1_finance <- subset ( ceosal1 , finance ==1)
ceosal1_consprod <- subset ( ceosal1 , consprod ==1)
ceosal1_utility <- subset ( ceosal1 , utility ==1)

modelo <- lm( salary ~ sales + roe + ros , data = ceosal1 )
modelo_indus <- lm( salary ~ sales + roe + ros , data = ceosal1_indus )
modelo_finance <- lm( salary ~ sales + roe + ros , data = ceosal1_finance )
modelo_consprod <- lm( salary ~ sales + roe + ros , data = ceosal1_consprod )
modelo_utility <- lm( salary ~ sales + roe + ros , data = ceosal1_utility )

SSR_r <- sum ( residuals ( modelo ) ^2)
SSR_indus <- sum ( residuals ( modelo_indus ) ^2)
SSR_finance <- sum ( residuals ( modelo_finance ) ^2)
SSR_consprod <- sum ( residuals ( modelo_consprod ) ^2)
SSR_utility <- sum ( residuals ( modelo_utility ) ^2)
SSR_ur <- SSR_indus + SSR_finance + SSR_consprod + SSR_utility

G <- 4
k <- 3
n <- nrow ( ceosal1 )
gl1 <- (G -1)*( k +1)

El test de Chow evalúa si los modelos de salario para los distintos tipos de empresa son iguales o presentan diferencias significativas. La hipótesis nula establece que todos los grupos tienen los mismos coeficientes, si el F-estadístico es menor que el F-crítico, no se rechaza la hipótesis nula, lo que indicaría que los modelos por tipo de empresa no son significativamente diferentes, es decir, la estructura salarial de los CEOs es similar entre los distintos sectores económicos analizados.